lamindb/core/storage/_zarr.py

```diff
@@ -48,29 +52,62 @@ def identify_zarr_type(
 
     try:
         storage = zarr.open(open_obj, mode="r")
-        if "spatialdata_attrs" in storage.attrs:
-            return "spatialdata"
         if storage.attrs.get("encoding-type", "") == "anndata":
             return "anndata"
+        elif storage.attrs.get("encoding-type", "") == "MuData":
+            return "mudata"
+        elif "spatialdata_attrs" in storage.attrs:
+            return "spatialdata"
     except Exception as error:
         logger.warning(f"an exception occured {error}")
     return "unknown"
 
 
-def load_anndata_zarr(storepath: UPathStr) -> AnnData:
+def load_zarr(
+    storepath: UPathStr,
+    expected_type: Literal["anndata", "mudata", "spatialdata"] = None,
+) -> ScverseDataStructures:
+    """Loads a zarr store and returns the corresponding scverse data structure.
+
+    Args:
+        storepath: Path to the zarr store
+        expected_type: If provided, ensures the zarr store is of this type ("anndata", "mudata", "spatialdata")
+                       and raises ValueError if it's not
+    """
     fs, storepath_str = infer_filesystem(storepath)
     if isinstance(fs, LocalFileSystem):
         # this is faster than through an fsspec mapper for local
         open_obj = storepath_str
     else:
         open_obj = create_mapper(fs, storepath_str, check=True)
-    adata = read_zarr(open_obj)
-    return adata
+
+    actual_type = identify_zarr_type(storepath)
```

Comment:
what i don't like here is that this makes a mapper (open_obj) again in addition to open_obj above. Maybe it can just return it?

Reply:
Ahh makes sense. I added an additional method that only create the `open_obj` and I am then reusing that.